### PR TITLE
Fix module resolution for preval.require

### PR DESCRIPTION
**Pull request by sergeybekrin**
**What**: `path.join` -> `path.resolve`

**Why**: `preval.require` won't resolve non-top level paths

**How**: by updating code which resolves absolute path to module

This PR fixes an issue then using `preval.require('../../foo');` for non-top level modules. This results in resolution from incorrect base parent directory and throws `Module not found` with actually correct (from project root perspective) path, which is super confusing.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -121,7 +119,7 @@ function prevalPlugin({types: t, template, transformFromAst}) {
           }
           return result.value
         })
-        const absolutePath = p.join(p.dirname(filename), source.node.value)
+        const absolutePath = p.resolve(p.dirname(filename), source.node.value)
```
Comment (sergeybekrin) [inline src/index.js]: This is actual change, rest of the stuff is formatted by prettier I believe

**Comment by kentcdodds** [comment]
Thanks! Could you add a test to verify that the bug you found has been fixed and make sure it doesn't come back? 👍

**Comment by sergeybekrin** [comment]
@kentcdodds will do 🙌 Meanwhile, can you suggest most suitable structure for that kind of test? Can't think of anything but creating nested directory in __tests__.

**Comment by kentcdodds** [comment]
You could do it in [fixtures](https://github.com/kentcdodds/babel-plugin-preval/tree/master/src/__tests__/fixtures) :+1:

**Comment by sergeybekrin** [comment]
That was quick! 👍

**Pull request merged**